contracts/HATVaults.sol

```diff
@@ -485,7 +486,9 @@ contract  HATVaults is Governable, ReentrancyGuard {
         }
         if (_amount > 0) { // will only be 0 in case of claimReward
             uint256 lpSupply = pool.balance;
+            uint256 balanceBefore = pool.lpToken.balanceOf(address(this));
             pool.lpToken.safeTransferFrom(address(msg.sender), address(this), _amount);
+            require(pool.lpToken.balanceOf(address(this)) - balanceBefore == _amount, "HVE45");
```

Comment:
we don't want to disable deposits in case there is a fee taken or something like that, just to check how much was actually transferred and update everything accordingly

Reply:
same for withdrawal

Reply:
exactly, replace this line with `_amount = pool.lpToken.balanceOf(address(this)) - balanceBefore` and I guess we are done

Reply:
NOT the same for withdrawal, though, with withdrawal I guess the logic can be left unchanged